Client Generalization Refactor (#94)
cosmos/sdk-application-tutorial: tutorial/queriers.md

# Queriers

## Query Types

Start by creating the `./x/nameservice/types/querier.go` file. This is where you will define your querier types.

```go
package types

import (
	"strings"
)

// Query Result Payload for a resolve query
type QueryResResolve struct {
	Value string `json:"value"`
}

// implement fmt.Stringer
func (r QueryResResolve) String() string {
	return r.Value
}

// Query Result Payload for a names query
type QueryResNames []string

// implement fmt.Stringer
func (n QueryResNames) String() string {
	return strings.Join(n[:], "\n")
}
```

## Querier

> Start by creating the `./x/nameservice/querier.go` file. This is the place to define which queries against application state users will be able to make. Your `nameservice` module will expose three queries:

Reviewer: ```suggestion
Now you can create the `./x/nameservice/querier.go` file. This is the place to define which queries against application state users will be able to make. Your `nameservice` module will expose three queries:
```
